app/main.py

```diff
@@ -1 +1,34 @@
-#  write your code here
+class Animal:
+    def __init__(self, name: str, appetite: int, is_hungry: bool = True):
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self):
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self):
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True):
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    def catch_mouse(self):
+        print("The hunt began!")
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True):
+        super().__init__(name, appetite=7, is_hungry=is_hungry)
+
+    def bring_slippers(self):
+        print("The slippers delivered!")
+
+
+def feed_animals(list_animals: list) -> int:
+    return sum(item.feed() for item in list_animals)
```

Comment:
Consider adding a type hint for the elements of `list_animals` to ensure they are instances of `Animal`. This can help with code readability and type checking. For example, you can use `list[Animal]` instead of `list`.